geomstats/geometry/lie_algebra.py

```diff
@@ -0,0 +1,147 @@
+    ])
+
+
+class MatrixLieAlgebra:
+    """Class implementing matrix Lie algebra related functions."""
+
+    def __init__(self, dimension, n):
+        """Construct the MatrixLieAlgebra object.
+
+        Parameters
+        ----------
+        dimension: int
+            The dimension of the Lie algebra as a real vector space
+        n: int
+            The amount of rows and columns in the matrx representation of the
+            Lie algebra
+        """
+        self.dimension = dimension
+        self.n = n
+        self.basis = None
+
+    def lie_bracket(self, matrix_a, matrix_b):
+        """Compute the Lie_bracket (commutator) of two matrices.
+
+        Notice that inputs have to be given in matrix form, no conversion
+        between basis and matrix representation is attempted.
+        Parameters
+        ----------
+        matrix_a: array-like, shape=[n_sample, n, n]
+        matrix_b: array-like, shape=[n_sample, n, n]
+
+        Returns
+        -------
+        bracket: shape=[n_sample, n, n]
+
+        """
+        return gs.matmul(matrix_a, matrix_b) - gs.matmul(matrix_b, matrix_a)
+
+    def baker_campbell_hausdorff(self, matrix_a, matrix_b, order=2):
+        """Calculate the Baker Campbell Hausdorff approximation of given order.
+
+        We use the algorithm published by Casas / Murua in their paper
+        "An efficient algorithm for computing the Baker–Campbell–Hausdorff
+        series and some of its applications" in J.o.Math.Physics 50 (2009) as
+        well as their computed constants from
+        http://www.ehu.eus/ccwmuura/research/bchHall20.dat.
+        Our file is truncated to enable us to calculate BCH up to order 15
+
+        This represents Z =log(exp(X)exp(Y)) as an infinite linear combination
+        of the form
+            Z = sum z_i E_i
+        where z_i are rational numbers and E_i are iterated Lie brackets
+        starting with E_1 = X, E_2 = Y, each E_i is given by some i',i'':
+            E_i = [E_i', E_i''].
+
+        Parameters
+        ----------
+        matrix_a: array-like, shape=[n_sample, n, n]
+        matrix_b: array-like, shape=[n_sample, n, n]
+        order: int
+            the order to which the approximation is calculated. Note that this
+            is NOT the same as using only E_i with i < order
+        """
+        if order > 15:
+            raise NotImplementedError("BCH is not implemented for order > 15.")
+
+        number_of_hom_degree = gs.array(
+            [
+                2,
+                1,
+                2,
+                3,
+                6,
+                9,
+                18,
+                30,
+                56,
+                99,
+                186,
+                335,
+                630,
+                1161,
+                2182
+            ]
+        )
+        n_terms = gs.sum(number_of_hom_degree[:order])
+
+        Ei = gs.zeros((n_terms, self.n, self.n))
```

Comment:
no uppercase letters in names that are not classes or constant as per PEP8 guidelines